analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -233,14 +242,35 @@ class Cargo(
             { extractPackage(it, hashes) }
         )
 
-        val dependencies = resolveDependencyTree(projectName, projectVersion, metadata, packages) { id, devId ->
-            !isDevDependencyOf(id, devId, metadata)
+        val projectId = metadata["workspace_members"]
+            .map { it.textValueOrEmpty() }
+            .single { it.contains(projectName) && it.contains(projectVersion) }
+
+        val projectNode = metadata["packages"].single { it["id"].textValueOrEmpty() == projectId }
+
+        val directDependencies = projectNode["dependencies"].filter { node ->
+            val kind = node["kind"].textValueOrEmpty()
+            kind != "dev" && kind != "build"
+        }.mapNotNull {
+            val name = it["name"].textValue()
+            val version = getResolvedVersion(projectName, projectVersion, it["name"].textValue(), metadata)
+            if (version == null) null else Pair(name, version)
```

Comment:
Shorter: `Pair(name, version).takeUnless { version == null }`.

Reply:
It's not shorter but exactly the same length. And it creates the `Pair` even if it is not required. And Kotlin infers the type to `Pair<String, String?>` instead of `Pair<String, String>` this way.